### PR TITLE
clair: limit db connections pool (PROJQUAY-2659)

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Restrict the max number of open connections to the clair postgres. If
HPA is managed we have control over the max number of pods so we can
calculate the max number of connections per pod. If HPA is not managed
then we can't properly estimate the max number of pods so we decrease
the connection pool size.

### CODE REVIEW

File: pkg/kustomize/secrets.go [hunk 1]
```diff
@@ -406,9 +416,8 @@ func clairConfigFor(log logr.Logger, quay *v1.QuayRegistry, quayHostname, preSha
 			"migrations":             true,
 		},
 		"matcher": map[string]interface{}{
-			"connstring":    dbConn,
-			"max_conn_pool": 100,
```
Comment (dmage) [inline pkg/kustomize/secrets.go:410]: How do we decide what should be in this map and what should be in the connstring string?

Comment (ricardomaraschini) [reply pkg/kustomize/secrets.go:410]: @quay/clair should be able to shine some light here

Comment (hdonnay) [reply pkg/kustomize/secrets.go:410]: Clair's config is documented (via the go types) [here](https://pkg.go.dev/github.com/quay/clair/config#Config). In this specific instance, we [deprecated](https://pkg.go.dev/github.com/quay/clair/config#Matcher) the dedicated connection pool size to make it uniformly specified in the connection string.

**Review by dmage — APPROVED** [review]

**Pull request merged**